File: week6/old-examples/javascript/src/domain/geometry/shapes/Rectangle.js
import { Shape } from "../contracts/Shape.js";

export class Rectangle extends Shape {
  constructor(width, height) {
    super();
    this.width = width;
    this.height = height;
  }
  area() { return this.width * this.height; }
  perimeter() { return 2 * this.width + 2 * this.height; }
}

File: week6/old-examples/csharp/src/OopOrganization.Chapter7/Domain/Geometry/Shapes/Rectangle.cs
using OopOrganization.Domain.Geometry.Contracts;

namespace OopOrganization.Domain.Geometry.Shapes;

public sealed class Rectangle : IShape
{
    public double Width { get; }
    public double Height { get; }

    public Rectangle(double width, double height)
    {
        Width = width;
        Height = height;
    }

    public double Area() => Width * Height;

    public double Perimeter() => 2 * Width + 2 * Height;
}
